apps/docs/pages/guides/auth.mdx

```diff
@@ -84,7 +84,7 @@ For deployments with Vercel, set the `SITE_URL` to your official site URL. Add t
 - `http://localhost:3000/**`
 - `https://*-username.vercel.app/**`
 
-Vercel provides an environment variable for the URL of the deployment called `NEXT_PUBLIC_VERCEL_URL`. See the [Vercel docs](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) for more details. You can use this variable to dynamically redirect depending on the environment:
+Vercel provides an environment variable for the URL of the deployment called `NEXT_PUBLIC_VERCEL_URL`. See the [Vercel docs](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) for more details. You can use this variable to dynamically redirect depending on the environment. You should also set the value of the environment variable called NEXT_PUBLIC_SITE_URL, this should be set to your site URL in production environment to ensure that redirects function correctly.
```

Comment:
```suggestion
Vercel provides a built-in environment variable for the URL of the deployment called `NEXT_PUBLIC_VERCEL_URL`. See the [Vercel docs](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) for more details. You can use this variable to dynamically redirect depending on the environment. You may also want to consider creating a custom environment variable called `NEXT_PUBLIC_SITE_URL` that contains your production site URL, as sometimes this is different than `NEXT_PUBLIC_VERCEL_URL` (eg. when using custom domains).
```

Reply:
Hi @gregnr 

On a slightly related note, setting up a local dev workflow has an esoteric glitch. It relates to NEXT_PUBLIC_SITE_URL 

When you use the Vercel CLI to get the local environment variables `vercel env pull .env.local`, you find that within .env.local, NEXT_PUBLIC_SITE_URL defaults to the production primary domain 🤔.  So even when running locally the user is redirected to the prod site on Vercel after authentication

This is fixed by editing env.local and setting NEXT_PUBLIC_SITE_URL="http://localhost:3000"

Setting it to localhost keeps the entire app flow local including after authentication

I don't know where it's best to record the knowledge but wanted to share as I've noticed a few posts across multiple forums by folks with this problem.

cheers,
Andrew.

Reply:
Good mention @abwdev - I think the idea is to set up multiple `NEXT_PUBLIC_SITE_URL` for each environment on Vercel. So set `NEXT_PUBLIC_SITE_URL=http://localhost:3000` for dev and `NEXT_PUBLIC_SITE_URL=<prod-domain>` on prod, etc.

Reply:
Thank you @gregnr that helps a lot!